--- src/tools/moc/moc.cpp.orig	2012-04-26 21:46:08.000000000 +0200
+++ src/tools/moc/moc.cpp	2012-08-12 12:31:12.411828701 +0200
@@ -777,8 +777,8 @@
         fn = filename.mid(i);
     fprintf(out, "/****************************************************************************\n"
             "** Meta object code from reading C++ file '%s'\n**\n" , (const char*)fn);
-    fprintf(out, "** Created: %s\n"
-            "**      by: The Qt Meta Object Compiler version %d (Qt %s)\n**\n" , dstr.data(), mocOutputRevision, QT_VERSION_STR);
+    fprintf(out, "** Created\n"
+            "**      by: The Qt Meta Object Compiler version %d (Qt %s)\n**\n" , mocOutputRevision, QT_VERSION_STR);
     fprintf(out, "** WARNING! All changes made in this file will be lost!\n"
             "*****************************************************************************/\n\n");
 
--- src/tools/uic/uic.cpp.orig	2012-04-26 21:46:08.000000000 +0200
+++ src/tools/uic/uic.cpp	2012-08-12 12:35:27.502830993 +0200
@@ -139,7 +139,7 @@
         out << "/********************************************************************************\n";
         out << "** Form generated from reading UI file '" << QFileInfo(opt.inputFile).fileName() << "'\n";
         out << "**\n";
-        out << "** Created: " << QDateTime::currentDateTime().toString() << "\n";
+        out << "** Created" << "\n";
         out << "**      " << QString::fromLatin1("by: Qt User Interface Compiler version %1\n").arg(QLatin1String(QT_VERSION_STR));
         out << "**\n";
         out << "** WARNING! All changes made in this file will be lost when recompiling UI file!\n";
--- src/tools/rcc/rcc.cpp.orig	2012-04-26 21:46:08.000000000 +0200
+++ src/tools/rcc/rcc.cpp	2012-08-12 12:38:55.148832886 +0200
@@ -732,7 +732,6 @@
         writeString("** Resource object code\n");
         writeString("**\n");
         writeString("** Created: ");
-        writeByteArray(QDateTime::currentDateTime().toString().toLatin1());
         writeString("\n**      by: The Resource Compiler for Qt version ");
         writeByteArray(QT_VERSION_STR);
         writeString("\n**\n");
