--- mkspecs/common/gcc-base.conf.orig	2012-04-26 21:45:52.000000000 +0200
+++ mkspecs/common/gcc-base.conf	2012-07-23 21:14:35.495641264 +0200
@@ -35,7 +35,7 @@
 QMAKE_CFLAGS_DEPS           += -M
 QMAKE_CFLAGS_WARN_ON        += -Wall -W
 QMAKE_CFLAGS_WARN_OFF       += -w
-QMAKE_CFLAGS_RELEASE        += -O2
+QMAKE_CFLAGS_RELEASE        += -O3
 QMAKE_CFLAGS_DEBUG          += -g
 QMAKE_CFLAGS_SHLIB          += -fPIC
 QMAKE_CFLAGS_STATIC_LIB     += -fPIC
--- mkspecs/common/g++-base.conf.orig	2012-04-26 21:45:52.000000000 +0200
+++ mkspecs/common/g++-base.conf	2012-07-23 21:15:40.937641856 +0200
@@ -13,7 +13,7 @@
 QMAKE_LINK_C       = $$QMAKE_CC
 QMAKE_LINK_C_SHLIB = $$QMAKE_CC
 
-QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO += -O2 -g
+QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO += -O3 -g
 
 QMAKE_CXX = g++
 
